src/services/BavService.ts

```diff
@@ -240,15 +240,16 @@ export class BavService {
 		}
 	}
 
-	async saveCopCheckResult(sessionId: string, copCheckResult: CopCheckResult): Promise<void> {
+	async saveCopCheckResult(sessionId: string, copCheckResult: CopCheckResult, retryCount?: number): Promise<void> {
 		this.logger.info({ message: "Saving session table with copCheckResult", copCheckResult });
 
 		const updateStateCommand = new UpdateCommand({
 			TableName: this.tableName,
 			Key: { sessionId },
-			UpdateExpression: "SET copCheckResult = :copCheckResult, authSessionState = :authSessionState",
+			UpdateExpression: `SET copCheckResult = :copCheckResult, authSessionState = :authSessionState${retryCount ? ", retryCount = :retryCount" : ""}`,
```

Comment:
`SET copCheckResult = :copCheckResult, authSessionState = :authSessionState, ${retryCount ? ", retryCount = :retryCount" : ""}` ?

Reply:
Sorry I can't see the difference apart from `authSessionState = :authSessionState,` having a comma but I removed that so that it's only appended of we need to add `retryCount` to the expression

Reply:
The confused the heck out of me sorry but yeah Ig et it now, if retryCount then you're adding `", retryCount = :retryCount"`